ipaserver/install/server/replicainstall.py

```diff
@@ -930,6 +930,9 @@ def promote_check(installer):
     installutils.verify_fqdn(config.master_host_name, options.no_host_dns,
                              local_hostname=not container_environment)
 
+    if config.host_name.lower() == options.domain_name.lower():
```

Comment:
This line raises an Exception when using the client promotion mecanism with `ipa-client-install` followed by `ipa-replica-install` because options.domain_name is not set:
```
# ipa-replica-install -U
Your system may be partly configured.
Run /usr/sbin/ipa-server-install --uninstall to clean up.

'NoneType' object has no attribute 'lower'
The ipa-replica-install command failed. See /var/log/ipareplica-install.log for more information
```
You can use config.domain_name.lower() instead.

Reply:
Bah, got it. I think I only tested a straight replica install. Fixed.

I added a specfiic test for replica promotion to cover this case.